### ISSUE TITLE
When does override-content-type apply?

### ISSUE THREAD

**Issue by ndw** [xproc/3.0-steps#174]
Consider this step:

```
<p:text-join suffix="Injected text." override-content-type="text/special">
  <p:with-input>
    <p:inline content-type="text/plain">Text</p:inline>
  </p:with-input>
</p:text-join>
```

The `result` port on `p:text-join` claims that it produces `text/plain`. But in this case it produces `text/special`. I think that should throw an error. (See also https://github.com/xproc/3.0-specification/issues/854)

I think we could make the output content type `text/*` to avoid this. Or we could remove the output constraint and let text join produce anything at all.

**Comment by ndw** [comment]
The error is `err:XD0042` per Achim's clarification on xproc/3.0-specification#854.

**Comment by xml-project** [comment]
I think text/* would be ok. But I am not sure I want XD0042 for, say:
````
<p:text-join suffix="Injected text." override-content-type="image/jpeg">
  <p:with-input>
    <p:inline content-type="text/plain">Text</p:inline>
  </p:with-input>
</p:text-join>
````
I would like to see a step error, saying that over-content-type is not compatible with "text/*".

**Comment by ndw** [comment]
Come to think of it, do we really need `override-content-type` on `p:text-join`? We have `p:cast-content-type` to do this kind of thing and if we have it on `p:text-join` why not on all of the steps? Is there a specific text-related use case that I'm failing to remember?

**Comment by xml-project** [comment]
I think @eriksiegel 's point was, that I might join text document all having say "text/csv" and I want the result to have this too.
This could be done by
````
<p:text-join /> <!-- producing text/plain as default -->
<p:cast-content-type /> <!-- change back to text/csv -->
<p:
````
Seems a bit odd, doesn't it?

**Comment by ndw** [comment]
I suppose. I doubt it’s unique in that regard, but I won’t fuss.

I propose we change the output content type to `text/*`

**Comment by xml-project** [comment]
Fixed with pr #177
